### PR TITLE
Feature magic lines v2

### PR DESCRIPTION
**Pull request by dmforeacre**
Major revamp of magic lines

### CODE REVIEW

File: UMLEditor/UMLEditor/Views/MainWindow.axaml.cs [hunk 1]
```diff
@@ -834,6 +878,16 @@ private void ViewEditToggle_OnClick(object sender, RoutedEventArgs e)
             #pragma warning restore CS8629
         }
 
+        private void MagicLinesToggle_OnClick(object sender, RoutedEventArgs e)
+        {
+            ToggleSwitch viewSwitch = (ToggleSwitch) sender;
+
+            RelationshipLine.ToggleMagicLines(((bool)viewSwitch.IsChecked));
+
```
Comment (Robunnin) [inline UMLEditor/UMLEditor/Views/MainWindow.axaml.cs:886]: We should be looking at cleaning up the warnings. More warnings have been introduced than were before

File: UMLEditor/UMLEditor/Views/ClassBox.axaml.cs [hunk 2]
```diff
@@ -169,8 +172,9 @@ public ClassBox()
                 // Set the ClassBox's location to the location of the cursor
                 Canvas.SetLeft(this, newX);
                 Canvas.SetTop(this, newY);
-
-                _parentWindow!.RedrawLines();
+                
+                // ****** COMMENT OUT THIS LINE FOR BETTER DRAW PERFORMANCE ****** 
+                //_parentWindow!.RedrawLines();
```
Comment (Robunnin) [inline UMLEditor/UMLEditor/Views/ClassBox.axaml.cs:177]: If this is going to be commented out you can just delete it.

**Comment by Robunnin** [comment]
**Loading a diagram causes an error to be thrown with no lines drawn.** 
image

**Toggling magic lines crashes the program**
image

**Comment by Robunnin** [comment]
**Magic lines don't avoid classboxes**
image

**Comment by Robunnin** [comment]
I don't really want a magic lines toggle, but if we are going to have it the window size needs to be adjusted so the editor is usable at lower resolutions.
image

**Comment by Robunnin** [comment]
Performance seems to be a bit better on average, but is still a big issue. I still have freezes that make it difficult to use the editor. Edge selection is also a bit wonky but still seems to be better. 

This branch needs to be updated with the new changes that were merged this morning.

**Review by Robunnin — DISMISSED** [review]
An improvement for sure, but there are still issues that make it hard to use the editor.

File: UMLEditor/UMLEditor/Views/MainWindow.axaml.cs (hunk 1 above)
Comment (dmforeacre) [reply UMLEditor/UMLEditor/Views/MainWindow.axaml.cs:931]: Warnings have been taken care of.

**Review by mayoung00 — APPROVED** [review]

**Pull request merged**